### ISSUE TITLE
Too many characters.

### ISSUE THREAD

**Issue by wayneluke** [tfmonk/Accountant#4]
I have 15 characters in Accountant and it won't properly total the amount of gold on the All Chars tab. 

**To Reproduce**
Install Accountant 10.0.0.0.0 (i've tried the Curseforge and Wago versions)
Log into the game. 
I get this message multiple times in the chat window:

```
Acc: ERROR: Too many characters saved. Please delete some in options!!  
Loaded Accountant 10.0.0.0.0 by tfmonk for 'Wyrmrest Accord-Jems' 
Acc: ERROR: Too many characters saved. Please delete some in options!!  
Acc: ERROR: Too many characters saved. Please delete some in options!!  
Acc: ERROR: Too many characters saved. Please delete some in options!!  
Acc: ERROR: Too many characters saved. Please delete some in options!!  
Acc: ERROR: Too many characters saved. Please delete some in options!!  
```

**Expected behavior**
I expect it to support up to 50 characters.

**Screenshots**

**Addon version (please complete the following information):**
 - 10.0.0.0.0

**Comment by tfmonk** [comment]
@wayneluke would you be able to log into the game and run the below and tell me what number it returns? 

`/run charcount = 0;for i,v in pairs(Accountant_SaveData) do charcount = charcount + 1 ;end;print(charcount)`

If it returns 50 then that's currently the max supported amount, then I would recommend going into the options menu by right clicking the Accountant mini map icon or Options > AddOns > Accountant and going to the Remove Character sub menu and removing any old characters/trail ones you may of made which are logged in accountant.

It should be able to handle 50 characters across all realms, i haven't updated it support 60 yet which is the new max character count as that's a recent change

**Comment by wayneluke** [comment]
It returns the value of 15.

https://imgur.com/a/g27jA3u

https://imgur.com/a/oKCrQr8

**Comment by tfmonk** [comment]
thanks, I've found the issue and will need to look into it, looks likes its some old code to limit 15 characters per server but that was changed in 8.1.5, to allow 50 (now 60) all on one realm

**Comment by Evilvonbek** [comment]
> thanks, I've found the issue and will need to look into it, looks likes its some old code to limit 15 characters per server but that was changed in 8.1.5, to allow 50 (now 60) all on one realm

came to add it as issue but this one is open.

Can conform after doing some prune/recreate for heritage sets, it breaks at 16 on a single server so will be down to old legacy limits and quite probably blizz code vs anything wrong in addon. :D

**Comment by tfmonk** [comment]
I've found a way to get it not to break and it will display the total gold correctly but it will only display 15 characters still for now, this fix should be available later today

**Comment by Evilvonbek** [comment]
> I've found a way to get it not to break and it will display the total gold correctly but it will only display 15 characters still for now, this fix should be available later today

just to confirm, total gold + revenue figures etc will be correct but the char list will truncate at first 15 on server?

even more than hoped for if so, and loving the addon so far anyway as it's letting me keep a decent track of what's actually eating or making me cash rather than trying to remember if something sold or not XD

**Comment by tfmonk** [comment]
yeah gold+revenue figures will be correct no data is lost at all, I'm just changing it to say if character is 16 or above do not display on the all_char tab... I'm in the process of uploading an alpha version on curse to validate

**Comment by tfmonk** [comment]
@wayneluke @Evilvonbek the new version should be live on wago and curseforge now, I will try continue to look at a way of adding a scroll functionality so all characters can be displayed.
